src/arch/host/include/arch/string.h

```diff
@@ -32,7 +32,53 @@
 #ifndef __INCLUDE_ARCH_STRING_SOF__
 #define __INCLUDE_ARCH_STRING_SOF__
 
+#include <errno.h>
+
 #define arch_memcpy(dest, src, size) \
 	memcpy(dest, src, size)
 
+#define arch_bzero(ptr, size) \
+	memset(ptr, 0, size)
+
+
+void *memcpy(void *dest, const void *src, size_t length);
+void *memset(void *dest, int data, size_t count);
+int memset_s(void *dest, size_t dest_size,
+	     int data, size_t count);
+int memcpy_s(void *dest, size_t dest_size,
+	     const void *src, size_t src_size);
+
+static inline int arch_memcpy_s(void *dest, size_t dest_size,
+				const void *src, size_t src_size)
+{
+	if (!dest || !src)
+		return -EINVAL;
+
+	if ((dest + dest_size >= src && dest + dest_size <= src + src_size) ||
+		(src + src_size >= dest && src + src_size <= dest + dest_size))
+		return -EINVAL;
+
+	if (src_size > dest_size)
+		return -EINVAL;
```

Comment:
Should this be `return EINVAL;`? See discussion in PR #2644

Reply:
No, EINVAL is a positive value. Errors must be returned as negative, thus -EINVAL is correct.